hasktorch/src/Torch/Typed/Factories.hs

```diff
@@ -36,6 +36,18 @@ import Torch.Typed.Aux
 import Torch.Typed.Tensor
 import Prelude hiding (sin)
 
+instance Semigroup (Tensor device dtype shape) where
+  (<>) a b = UnsafeMkTensor $ D.add (toDynamic a) (toDynamic b)
+
+instance (TensorOptions shape dtype device) => Monoid (Tensor device dtype shape) where
```

Comment:
why do you need this concrete monoid for Tensor and NamedTensor?
Perhaps consider https://hackage.haskell.org/package/base-4.14.1.0/docs/Data-Monoid.html#v:Sum or https://hackage.haskell.org/package/base-4.14.1.0/docs/Data-Monoid.html#v:Product from base.

Reply:
I need a default constructor like zeros for this code.
https://github.com/hasktorch/hasktorch/pull/536/files/2ab534287bd778c9325ead060f1a878fde0453da#diff-ed8556e187a95c50a009496221d2e54ca9fe54ceafd52bbabd8470f10c1677fbR69-R79

Is it better to have an instance of Data.Default?